metaseq/sequence_generator.py

```diff
@@ -254,8 +254,6 @@ def _generate(
                 model_out, log_probs=True, sample=None
             )
             lprobs = lprobs[:, -1, :]
-            if self.need_logprobs:
```

Comment:
Conditionally seem to be missing the branch of `step >= max_len and self.need_logprobs == True` now - is that intended?

Reply:
yes. the step == max_len step is forced to be an EOS token regardless. So we don't need to record its score into the logprobs (that's also why scores is protected from writing)